git-tools/git-change-log

```diff
@@ -194,10 +195,14 @@ class git_change_log(object):
                         print("Ignore:", commit)
                         continue
 
+                    upstream_commit_hash = match.group(1)
                     try:
-                        upstream_commit = self.UpstreamRepo.commit(match.group(1))
-                    except Exception as e:
-                        print("Ignored (referenced commit not found in upstream):", commit)
+                        upstream_commit = self.UpstreamRepo.commit(upstream_commit_hash)
+                    except ValueError as e:
+                        print(f"Error: Commit hash {upstream_commit_hash}\n found with regex: {regex}\n referenced by commit {commit.hexsha} not found in upstream")
+                        continue
+                    except BadName as e:
+                        print(f"Error: Commit hash {upstream_commit_hash}\n found with regex: {regex}\n referenced by commit {commit.hexsha} is not a valid commit hash (maybe an issue with the regex?)")
                         continue
```

Comment:
Please squash both commits to one change and update the commit message accordingly